### PR TITLE
Upload Spotify's internal changes

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Upload Spotify's internal changes. 

## Type
 - [X] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
In Spotify we started the development of flytekit-java as a private repo and from time to time we "exported" our changes to the public repository. This was easy for us to start but of course is not how it is meant to be, this is our the last code upload before we close our internal repository, after this(*) our development will be in the open.

The changes included in this PR are the following:
### Bug fixes
* Fix bug with duplicated subworkflows
* Fix Remove SdkBranchNode#getSubNode as it is not supported in flytepropeller v0.12.9
* Fix maven settings for incremental builds
* Fix nested subworkflow serialization
* Fix Check for subworkflow cycles . 
* Ensure the range of Datetime values matches the range for [Timestamp in protoduf](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp)
* Verify node ids conforms with format 
* Give "Not supported error" in unit testing for branch or subworkflow nodes
* Write Container error when there are problems reading `input.pb` for the task.

### New features
* Add scala_2.13 support
* Add integration tests
* Use taskTemplatePath instead of index file in JFlyte tasks
* Add Dynamic Workflows support
* Publish javadoc and sources classifiers for scala modules
* Switch to softwaremill magnolia from propensive magnolia in the Scala Sdk. The later one is not supported anymore.
* Add node metadata overrides for names and timeouts.
* Add Resources support

## Follow-up issue
_NA_

### CODE REVIEW

**Review by veggiemonk — APPROVED** [review]

**Review by ckiosidis — APPROVED** [review]
I took a quick look through the files and they they seem alright
